canonical-web-and-design/maas-ui
# Build PoolSelect component for KVM compose form.

## Done

- Built PoolSelect component to show finer storage pool details in the KVM compose form.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM details page of a KVM with multiple storage pools. Ideally not an LXD KVM so you can add multiple disks.
- Add one or more disks then open the "Location" selects
- Check that the entered disk sizes match the requested portion of the meter
- Check that changing the disk size updates the "Requested" and "Free" portions of the meter correctly
- Check that changing the disk size to higher than a pool's available storage disables the pool, and if it was already selected, disables the form
- Check that changing the selected pool and deleting disks still work as expected
- Check that composing a machine still works correctly

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2066

## Screenshot
![2020-08-05_17-35](https://user-images.githubusercontent.com/25733845/89385977-ca989780-d743-11ea-9e05-9f3775504cef.png)
